@kiva/kv-components/vue/KvTextLink.vue

```diff
@@ -73,44 +80,59 @@ export default {
 			},
 		},
 	},
-	computed: {
-		tag() {
-			if (this.to) {
+	emits: [
+		'click',
+	],
+	setup(props, { emit }) {
+		const {
+			to,
+			href,
+		} = toRefs(props);
+
+		const buttonRef = ref(null);
+
+		const tag = computed(() => {
+			if (to.value) {
 				return 'router-link';
 			}
-			if (this.href) {
+			if (href.value) {
 				return 'a';
 			}
 			return 'button';
-		},
-	},
-	watch: { href() { this.setHref(); } },
-	mounted() {
-		this.setHref();
-	},
-	methods: {
-		onClick(event) {
+		});
+
+		const onClick = (event) => {
 			// emit a vue event and prevent native event
 			// so we don't have to write @click.native in our templates
-			if (this.tag === 'button') {
+			if (tag === 'button') {
```

Comment:
I think this needs to check `tag.value` instead of `tag`

Reply:
Fixed, thanks!